SmartDeviceLinkTests/DevAPISpecs/SDLLockScreenManagerSpec.m

```diff
@@ -433,11 +433,12 @@ @interface SDLLockScreenManager ()
 
             it(@"should present the lock screen if not already presented", ^{
                 OCMStub([mockViewControllerPresenter lockViewController]).andReturn([OCMArg any]);
-                OCMStub([mockViewControllerPresenter presented]).andReturn(false);
 
                 [[NSNotificationCenter defaultCenter] postNotification:testLockStatusNotification];
 
-                OCMVerify([mockViewControllerPresenter present]);
+                // Since lock screen must be presented/dismissed on the main thread, force the test to execute manually on the main thread. If this is not done, the test case may fail.
+                [[NSRunLoop mainRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.01]];
```

Comment:
Does `dispatch_sync` work for this case?

Reply:
I tried it and it crashed.

Reply:
Is that because it's already on the main queue?